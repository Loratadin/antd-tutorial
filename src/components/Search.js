import React, { PureComponent } from 'react';
import { Select, Button } from "antd";
import { makesAndModels, bodyStyles } from "../dummyData";

const { Option } = Select;
export default class Search extends PureComponent {
    state = {
        byMake: true,
        byStyle: false,
        selectedMake: undefined,
        selectedModel: undefined,
        selectedBodyStyle: undefined,
    }

    updateSelectedOption = (value, key) => this.setState({ [key]: value });

    switchTab = (key1, key2) => {
        if (this.state[key1] === false) {
            this.setState({[key1]: true, [key2]: false})
        }
    }
    onSearchClick = () => {
        this.props.changeTab("cars");
    }

    onSeeAllClick = () => {
        this.props.changeTab("cars");
    }

    render() {
        const { byMake, byStyle, selectedMake, selectedModel, selectedBodyStyle } = this.state;
        const models = selectedMake && makesAndModels.find(record => record.make === selectedMake).models;
        return (
            <div data-testid="search">
                <div className="search__tabs">
                    <div
                        className={`search__tab ${byMake && "search__tab-active"}`}
                        onClick={() => this.switchTab("byMake", "byStyle")}
                    >
                        Search By Make
                    </div>
                    <div
                        className={`search__tab ${byStyle && "search__tab-active"}`}
                        onClick={() => this.switchTab("byStyle", "byMake")}
                    >
                        Search By Body Style
                    </div>
                </div>
                <div className="search__body">
                    {byMake ? (
                        <div className="search__selector-wrap">
                            <Select
                                className="search__selector"
                                placeholder="Select a Make"
                                value={selectedMake}
                                onChange={(value => this.updateSelectedOption(value, "selectedMake"))}
                            >
                                {makesAndModels.map(item => (
                                    <Option key={item.make} value={item.make}>
                                        {item.make}
                                    </Option>
                                ))}
                            </Select>
                            <Select
                                className="search__selector"
                                placeholder="Select a Model"
                                value={selectedModel}
                                onChange={(value => this.updateSelectedOption(value, "selectedModel"))}
                            >
                                {selectedMake && models.map(model => (
                                    <Option key={model} value={model}>
                                        {model}
                                    </Option>
                                ))}
                            </Select>
                        </div>
                    ) : (
                        <div className="search__selector-wrap">
                            <Select
                                className="search__selector"
                                placeholder="Select a Body Style"
                                value={selectedBodyStyle}
                                onChange={(value => this.updateSelectedOption(value, "selectedBodyStyle"))}
                            >
                                {bodyStyles.map(item => (
                                    <Option key={item} value={item}>
                                        {item}
                                    </Option>
                                ))}
                            </Select>
                        </div>
                    )}
                    <Button type="primary" onClick={this.onSearchClick}>Search</Button>
                    <Button className="search__see-all-button" onClick={this.onSeeAllClick} data-testid="see-all">See All Cars</Button>
                </div>
            </div>
        );
    }
}