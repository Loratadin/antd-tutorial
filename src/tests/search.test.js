import React from 'react';
import ReactDOM from 'react-dom';
import Search from '../components/Search';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Search/>, div)
})
it("renders button correctly", () => {
    const { getByTestId } = render(<Search />)
    expect(getByTestId('see-all')).toHaveTextContent("See All Cars")
})
it("renders search component correctly", () => {
    const { getByTestId } = render(<Search />)
    expect(getByTestId('search')).toHaveTextContent("Search")
})
it("matches snapshot", () => {
    const tree = renderer.create(<Search />).toJSON();
    expect(tree).toMatchSnapshot();
})