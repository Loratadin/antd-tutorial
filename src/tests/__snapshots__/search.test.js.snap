// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="search"
>
  <div
    className="search__tabs"
  >
    <div
      className="search__tab search__tab-active"
      onClick={[Function]}
    >
      Search By Make
    </div>
    <div
      className="search__tab false"
      onClick={[Function]}
    >
      Search By Body Style
    </div>
  </div>
  <div
    className="search__body"
  >
    <div
      className="search__selector-wrap"
    >
      <div
        className="search__selector ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          aria-autocomplete="list"
          aria-controls="test-uuid"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection__placeholder"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "display": "block",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              Select a Make
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <i
              aria-label="icon: down"
              className="anticon anticon-down ant-select-arrow-icon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
      </div>
      <div
        className="search__selector ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          aria-autocomplete="list"
          aria-controls="test-uuid"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection__placeholder"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "display": "block",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              Select aa Model
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <i
              aria-label="icon: down"
              className="anticon anticon-down ant-select-arrow-icon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
      </div>
    </div>
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      type="button"
    >
      <span>
        Search
      </span>
    </button>
    <button
      className="ant-btn search__see-all-button"
      data-testid="see-all"
      onClick={[Function]}
      type="button"
    >
      <span>
        See All Cars
      </span>
    </button>
  </div>
</div>
`;
